cogent/isa/Subtyping.thy

```diff
@@ -138,74 +138,32 @@ lemma type_lub_type_glb_commut:
   "K \<turnstile> c \<leftarrow> a \<squnion> b \<Longrightarrow> K \<turnstile> c \<leftarrow> b \<squnion> a"
   "K \<turnstile> c \<leftarrow> a \<sqinter> b \<Longrightarrow> K \<turnstile> c \<leftarrow> b \<sqinter> a"
 proof (induct rule: type_lub_type_glb.inducts)
-  case (lub_trecord ts ts1 ts2 s s2 s1)
-(*
+  case (lub_trecord K ts ts1 ts2 s s1 s2)
   then show ?case
-  proof (intro type_lub_type_glb.intros)
-    fix n t t1 t2 b b1 b2
-    assume
-      "(n, t, b) \<in> set ts"
-      "(n, t1, b1) \<in> set ts1"
-      "(n, t2, b2) \<in> set ts2"
-    then show "t \<leftarrow> t2 \<squnion> t1 \<and> b = inf b2 b1"
-      using lub_trecord
-      by (auto simp add: inf_commute)
-  qed simp+
+    apply (intro type_lub_type_glb.intros)
+    using list_all3_comm2 list_all3_mono apply fastforce
+         apply (induct rule: list_all3_induct)
+          apply simp
+    using sup_commute apply fastforce
+    by (simp+)[5]
```

Comment:
Avoid ending apply-style goals with by, as you loose the indent information on how many subgoals were left to prove, and it's mixing apply and Isar style.

Similarly with using, it's hard not to use using sometimes, but putting the apply on the line after the using indents the apply properly.